### PR TITLE
Immortal mode + refactoring

### PR DESCRIPTION
**Pull request by nima3333**
Three small changes:
- immortal mode implemented by bypassing `ed_fm_on_damage` if activated
- unlimited fuel name convention
- refactoring of drop tank switch logic to be more clear

### CODE REVIEW

File: A-4E-C/ExternalFM/FM/FuelSystem2.cpp [hunk 1]
```diff
@@ -97,11 +104,9 @@ void FuelSystem2::addFuel( double dm, bool wingFirst )
 			externalTanks += m_fuelCapacity[i] > 15.0;
 		}
 
-		double fracDM = dm / (double)externalTanks;
-
-
 		if ( externalTanks > 0 )
 		{
+			double fracDM = dm / (double)externalTanks;
```
Comment (08jne01) [inline A-4E-C/ExternalFM/FM/FuelSystem2.cpp]: Why did you move it inside the if?

Comment (08jne01) [reply A-4E-C/ExternalFM/FM/FuelSystem2.cpp]: There was a fix here relating to the external fuel tanks I can't remember the exact details though.

Comment (nima3333) [reply A-4E-C/ExternalFM/FM/FuelSystem2.cpp]: Even if very unlikely, this was to avoid a divby0 as it was not used outside this if.
Anyway, I messed up with this PR by modifying the branch after, I will resubmit later.
